File: bubble_sort.js
const numbers = [1, 32, 4, -12, 6, 8, 12, 6, 9, 7, 34, 65, 34, 89, 13, 56];
function bubble_sort(arr) {
    const s = performance.now()
    for (let i = 0; i < arr.length; i++) {
        for (let j = 0; j < arr.length; j++) {
            if (arr[j] > arr[j + 1]) {
                const temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
            }
        }
    }
    console.log('main', performance.now() - s);
    return arr;
}
bubble_sort(numbers)

File: bubble_sort.c
#include <stdio.h>
void bubble_sort(int arr[], int len);
void main()
{
    int array[5] = {3, 2, 4, 1, 5};
    bubble_sort(array, 5);
    for (int i = 0; i < 5; i++)
    {
        printf("%d\n", array[i]);
    }
}

void bubble_sort(int arr[], int len)
{
    int i, j;
    for (i = 0; i < len; i++)
    {
        for (j = 0; j < (len - 1) - i; j++)
        {
            if (arr[j] > arr[j + 1])
            {
                int temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
            }
        }
    }
}